hooks/lib/gcr-registry-provider.bash

```diff
@@ -0,0 +1,20 @@
+login() {
+  # Currently assume the use of docker-credential-gcr to manage AuthN transparently.
+  echo "Plugin currently assumes that docker-credential-gcr is on PATH and configured. See https://github.com/GoogleCloudPlatform/docker-credential-gcr#configuration-and-usage if later docker pull/push fail."
+}
+
+configure_registry_for_image_if_necessary() {
+  # GCR does not have a concept of a repository for images within a registry like ECR does.
+  echo ""
+}
+
+get_registry_url() {
+  if [[ -z "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_GCP_PROJECT:-}" ]]; then
+    log_fatal "gcp-project in plugin settings must have a value." 34
+  fi
+  if [[ -z "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_REGISTRY_HOSTNAME:-}" ]]; then
+    echoerr "registry-hostname had no value, defaulting to gcr.io"
+    BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_REGISTRY_HOSTNAME="gcr.io"
+  fi
+  echo "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_REGISTRY_HOSTNAME}/${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_GCP_PROJECT}/${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_ECR_NAME:-"$(get_default_image_name)"}"
```

Comment:
Ugh. Not for this PR, but we should definitely have a generic alias for this option as you've suggested.

Note that the README says that this is an ECR-specific option, so we should either update it _or_ drop this logic from here until we have that `image-name`.

Reply:
Let me know which way you'd prefer to go on this one ^ @petemounce

Apart from that, looks all ready to merge!

Reply:
I think I'd prefer to leave this as-is within this PR, and iterate afterwards.

Reply:
Actually, wait - I'm not sure I follow what you're suggesting. Please would you elaborate?

GCR URIs take the form `https://{host}/{gcp-project}/{container}:{tag} - so, actually, I think `container` is the equivalent to ECR's repository now that I think about this more freshly.

Reply:
> I think container is the equivalent to ECR's repository

Yep, looks that way to me too. Happy to iterate on it.